### DISCUSSION TITLE
kruise-controller-manager 内存使用量咨询

### DISCUSSION THREAD

**Discussion by steven6718** [openkruise/kruise#1086] [Q&A]
你好，最近在k8s集群中使用openkruise，主要使用了cloneset、sidecarset和containerrecreaterequests三个crd的功能。集群规模大概200 nodes，7000pod，100+ cloneset，400+ cloneset工作负责对应的pod。
部署kruise-controller-manager的时候，最开始配置了2G内存，最近发现controller-manager出现了OOM，因此曾大了内存配置，随着cloneset的增加在不断上涨，目前已经使用了4G内存，通过pprof查看allocs和heap的值一直是相等的，且在不断增大，想咨询下是否正常。相关环境如下：
k8s版本：1.18
openkruise版本：v1.1.0
crd只安装了：cloneset、sidecarset、containerrecreaterequests、imagepulljobs和nodeimages

**Comment by FillZpp** [comment]
@steven6718 kruise-controller-manager 会为 cloneset、pod、node 等资源建立 informer 缓存，因此内存占用确实会随着这些资源数量的扩大而增长。
另外，在我们即将发布的 v1.3.0 新版本中，有不少运行时的内存优化点，等最近几天发布后也可以升级版本看下效果如何。
